vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/Grid.java

```diff
@@ -3801,4 +3801,33 @@ public void setColumnOrder(List<Column<T>> columns) {
     private void fireColumnReorderEvent(List<Column<T>> columns) {
         fireEvent(new ColumnReorderEvent<>(this, false, columns));
     }
+
+    /**
+     * Scrolls to the given row index. Scrolls so that the row is shown at the
+     * start of the visible area whenever possible.
+     * 
+     * If the index parameter exceeds current item set size the grid will
+     * scroll to the end.
+     *
+     * @param index
+     *            zero based index of the item to scroll to in the current view.
+     */
+    public void scrollToIndex(int rowIndex) {
+        getElement().callJsFunction("scrollToIndex", rowIndex);
+    }
+
+    /**
+     * Scrolls to the beginning of the first data row.
+     */
+    public void scrollToStart() {
+        scrollToIndex(0);
+    }
+
+    /**
+     * Scrolls to the end of the last data row.
+     */
+    public void scrollToEnd() {
+        scrollToIndex(getElement().getProperty("size", 0));
```

Comment:
This might not work if a tree grid has items expanded. `size` is only the number of root-level items.

Reply:
Can I use `_effectiveSize` then? Or do you have another suggestion?

Reply:
That should work

Reply:
Had to do it a bit differently since for some reason getting the property value for `_effectiveSize` from the server side does not seem to work

Reply:
Had to do it like this: `getElement().executeJs("$0.scrollToIndex($0._effectiveSize)", this.getElement());` since the `_effectiveSize` property does not seem to available from the server side.